azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/maven/MavenSettingHelper.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.maven;
+
+import com.microsoft.azure.tools.auth.exception.AzureLoginException;
+import com.microsoft.azure.tools.auth.exception.InvalidConfigurationException;
+import com.microsoft.azure.tools.auth.exception.MavenDecryptException;
+import com.microsoft.azure.tools.auth.model.MavenAuthConfiguration;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.settings.Server;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import java.util.Objects;
+
+public class MavenSettingHelper {
+    /**
+     * Get auth configuration from serverId.
+     *
+     * @param session           the maven session
+     * @param settingsDecrypter the decrypter
+     * @param serverId          the server id
+     * @return the auth configuration
+     * @throws AzureLoginException when there are errors decrypting maven generated password
+     */
+    public static MavenAuthConfiguration getAuthConfigurationFromServer(MavenSession session, SettingsDecrypter settingsDecrypter, String serverId)
+            throws AzureLoginException {
+        if (StringUtils.isBlank(serverId)) {
+            throw new IllegalArgumentException("Parameter 'serverId' cannot be null or empty.");
+        }
+        final Server server = session.getSettings().getServer(serverId);
+        if (server == null) {
+            throw new InvalidConfigurationException(String.format("serverId '%s' cannot be found in maven settings.xml.", serverId));
+        }
+        final MavenAuthConfiguration configurationFromServer = new MavenAuthConfiguration();
+        final Xpp3Dom configuration = (Xpp3Dom) server.getConfiguration();
```

Comment:
⬆️ getRawConfiguration() from session and server.
⬇️ buildConfiguration() from raw configuration